### PR TITLE
Search: Handle pagination edge case with excluded post types

### PR DESCRIPTION
**Pull request by jsnmoon**
#### Changes proposed in this Pull Request:
* Overrides incorrect API response value for paginated queries with excluded post types. This can occur when paginating with excluded post types configured via the Customizer. 

#### Jetpack product discussion
p8oabR-DS-p2#comment-5144

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your site with a Jetpack Search subscription.
* If possible, override the site search ID (via mu-config.php) to `191431359`.
* Ensure that you have WooCommerce installed on your site.
* Through the customizer, exclude posts, pages, and attachments (media).
* Summon the search modal and set the search order to `oldest`; use an empty search query (`/?s=&sort=oldest`).
* Scroll to the bottom of the modal to load more results. Repeat this until all results have loaded. (Without this bugfix, the results will disappear following the final pagination API request.)

### CODE REVIEW

File: projects/plugins/jetpack/modules/search/instant-search/store/reducer/api.js [hunk 1]
```diff
@@ -58,6 +58,13 @@ export function response( state = {}, action ) {
 				};
 				newState.results = [ ...( 'results' in state ? state.results : [] ), ...newState.results ];
 			}
+
+			// NOTE: There's a bug in the API where paginating to the next page can return an empty response.
+			//       This is most likely caused by setting excluded post types. Since `response.total` is unreliable
+			//       in this instance, manually set it to the length of the results array.
+			if ( Array.isArray( newState.results ) && newState.results.length !== newState.total ) {
+				newState.total = newState.results.length;
```
Comment (kangzj) [inline projects/plugins/jetpack/modules/search/instant-search/store/reducer/api.js:66]: Should we always use newState.results.length as total? As you mentioned above, response.total is not reliable for queries excluding post types. I saw it changed from 18 to 13 and then 0 following the test instructions.

**Comment by kangzj** [comment]
Good job @jsnmoon! 👍 
Tested in Firefox and Safafi, it worked!

File: projects/plugins/jetpack/modules/search/instant-search/store/reducer/api.js (hunk 1 above)
Comment (jsnmoon) [reply projects/plugins/jetpack/modules/search/instant-search/store/reducer/api.js:66]: > Should we always use newState.results.length as total?

Yes; this is what's essentially being done within the conditional. If the new `results` value is not an array or if the total value already equals the new `results` length, then there's no need to update the total.

**Review by kangzj — APPROVED** [review]
Good job Jason 👍

**Review by jeherve — APPROVED** [review]

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-9.6` in 57ff08e1a3eb89260ad651cdfaf5e3a9465fdfcf

**Comment by jsnmoon** [comment]
🚢 r223999-wpcom!

**Pull request merged**